flink-streaming-java/src/test/java/org/apache/flink/streaming/api/DataStreamTest.java

```diff
@@ -1232,6 +1233,32 @@ public void testObjectArrayKeyRejection() {
 		testKeyRejection(keySelector, keyTypeInfo);
 	}
 
+	private enum TestEnum {
+		FOO, BAR
+	}
+
+	@Test
+	public void testEnumKeyRejection() {
+		KeySelector<Tuple2<TestEnum, String>, TestEnum> keySelector =
```

Comment:
can we reuse the function `testKeyRejection` for the newly added test?

Reply:
`testArrayKeyRejection` should be a better name. it only used for arrays and cannot be 
 used for enum type.